items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala

```diff
@@ -88,26 +76,99 @@ class SierraItemUpdater(sierraSource: SierraSource)(
     } yield accessConditions
 
   def updateItems(items: Seq[DisplayItem]): Future[Seq[DisplayItem]] = {
+
+    // item number -> item
     val itemMap = items.map { item =>
       SierraItemIdentifier.fromSourceIdentifier(item.identifiers.head) -> item
     } toMap
 
-    val accessConditions = for {
-      accessConditionsMap <- getAccessConditions(itemMap.keys.toSeq)
+    val staleItems = itemMap
+      .filter { case (_, item) => isStale(item) }
+
+    debug(
+      s"Asked to update items ${itemMap.keySet}, refreshing stale items ${staleItems.keySet}"
+    )
+
+    for {
+      accessConditions <- getUpdatedAccessConditions(staleItems.keys.toSeq)
+
+      updatedItems = itemMap.map {
+        case (sierraId, item) =>
+          accessConditions.get(sierraId) match {
+            case Some(updatedAc) => updateAccessCondition(item, updatedAc)
+            case None            => item
+          }
+      }
+    } yield updatedItems.toSeq
+  }
+
+  /** Given a series of item IDs, get the most up-to-date access condition
+    * information from Sierra.
+    *
+    */
+  private def getUpdatedAccessConditions(
+    itemIds: Seq[SierraItemNumber]
+  ): Future[Map[SierraItemNumber, AccessCondition]] =
+    itemIds match {
+      case Nil => Future.successful(Map())
+
+      case _ =>
+        for {
+          accessConditionsMap <- getAccessConditions(itemIds)
+
+          // It is possible for there to be a situation where Sierra does not know about
+          // an Item that is in the Catalogue API, but this situation should be very rare.
+          // For example an item has been deleted but the change has not yet propagated.
+          // In that case it gets method "NotRequestable".
+          missingItemIds = itemIds
+            .filterNot { accessConditionsMap.keySet.contains(_) }
+
+          missingItemsMap = missingItemIds
+            .map(_ -> AccessCondition(method = AccessMethod.NotRequestable))
+            .toMap
+        } yield accessConditionsMap ++ missingItemsMap
+    }
+
+  /** There are two cases we care about where the data in the catalogue API
+    * might be stale:
+    *
+    *   1) An item was on request for a user, but has been returned to the stores.
+    *      Another user could now request the item, but the catalogue API will
+    *      tell you it's temporarily unavailable.
+    *
+    *   2) An item is in the closed stores, and been requested by a user.
+    *      Another user can no longer request the item, but the catalogue API will
+    *      tell you it's available.
+    *
+    * In all other cases, we can use the access conditions in the catalogue API.
+    * There may be a small delay in updating an item's information, but this is
+    * relatively tolerable, and allows us to simplify the logic in this API for
+    * determining item status.
+    *
+    */
+  private def isStale(item: DisplayItem): Boolean = {
```

Comment:
Do you want to add a link to https://github.com/wellcomecollection/wellcomecollection.org/blob/fbec553332d061a6cdec5580c591ca810833e629/catalogue/webapp/components/PhysicalItems/PhysicalItems.tsx#L25-L28 in the comments for bonus points?